File: 0242-valid-anagram/0242-valid-anagram.cpp
#include <unordered_map>
#include <string>
using namespace std;

class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;

        unordered_map<char, int> a, b;

        for (char c : s) {
            a[c]++;
        }

        for (char c : t) {
            b[c]++;
        }

        return a == b;  // unordered_map supports == operator (C++20)
    }
};

File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        set1 = {}
        set2 = {}

        if len(s) != len(t):
            return False

        for i in s:
            set1[i] = set1.get(i,0)+1

        for i in t:
            set2[i] = set2.get(i,0)+1
            
        
        return set1 == set2

        